trl/trainer/sft_trainer.py

```diff
@@ -169,42 +224,9 @@ def __init__(
             dict_args.pop("push_to_hub_token")
             args = SFTConfig(**dict_args)
 
-        # Model
-        if args.model_init_kwargs is not None and not isinstance(model, str):
-            warnings.warn(
-                "You passed model_init_kwargs to the `SFTConfig`, but your model is already instantiated. "
-                "The `model_init_kwargs` will be ignored."
-            )
-        if isinstance(model, str):
-            model = self._create_model_from_path(model, args)
-
-        # PEFT configuration and model wrapping
-        if peft_config is not None:
-            model = self._prepare_peft_model(model, peft_config, args)
-
```

Comment:
This is moved down so that the user doesn't wait for the model to be loaded to get error if the pad token is not correctly specified